### PR TITLE
Tobin gh1030 sparse binding size

### PR DESCRIPTION
**Pull request by tobine**
A couple of updates to how sparse bindings are handled.

Need to get size of sparse bindings and also account for separate memory bound in sparse cases when we're dealing with the memory object itself.

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -9214,7 +9214,7 @@ static bool ValidateBarriers(const char *funcName, VkCommandBuffer cmdBuffer, ui
 
         auto buffer_node = getBufferNode(dev_data, mem_barrier->buffer);
         if (buffer_node) {
-            auto buffer_size = buffer_node->binding.size;
+            auto buffer_size = buffer_node->GetMemorySize();
             if (mem_barrier->offset >= buffer_size) {
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: Sum-of-binding-sizes doesn't make any sense for this check if you have sparse residency... for the sparse binding only level, it's OK I suppose.

Comment (tobine) [inline layers/core_validation.cpp]: Yeah, on further thought I don't like this.
I think instead we should just track size returned from Get*MemoryRequirements and check against that. Reworking.

**Comment by tobine** [comment]
Pushed an update to this branch using size from memory requirements.
Also added new CL to rename BUFFER_NODE->BUFFER_STATE which I missed in my previous rename sweep.

**Comment by tobine** [comment]
Replaced by #1168

**Pull request closed without merging**